alf/layers.py

```diff
@@ -1025,6 +1026,366 @@ def bias(self):
         return self._bias
 
 
+@gin.configurable
+class ParamFC(nn.Module):
+    def __init__(self,
+                 input_size,
+                 output_size,
+                 activation=torch.relu_,
+                 use_bias=True,
+                 kernel_initializer=None,
+                 kernel_init_gain=1.0,
+                 bias_init_value=0.0):
+        """A fully connected layer that does not maintain its own weight and bias,
+        but accepts both from users. If the given parameter (weight and bias) 
+        tensor has an extra batch dimension (first dimension), it performs
+        parallel FC operation. 
+
+        Args:
+            input_size (int): input size
+            output_size (int): output size
+            activation (torch.nn.functional):
+            use_bias (bool): whether use bias
+        """
+        super(ParamFC, self).__init__()
+
+        self._input_size = input_size
+        self._output_size = output_size
+        self._activation = activation
+        self._use_bias = use_bias
+        self._kernel_initializer = kernel_initializer
+        self._kernel_init_gain = kernel_init_gain
+        self._bias_init_value = bias_init_value
+
+        self._weight_length = output_size * input_size
+        self.set_weight(torch.randn(1, self._weight_length))
+        if use_bias:
+            self._bias_length = output_size
+            self.set_bias(torch.randn(1, self._bias_length))
+        else:
+            self._bias_length = 0
+            self._bias = None
+
+    @property
+    def weight(self):
+        """Get stored weight tensor or batch of weight tensors."""
+        return self._weight
+
+    @property
+    def bias(self):
+        """Get stored bias tensor or batch of bias tensors."""
+        return self._bias
+
+    @property
+    def weight_length(self):
+        """Get the n_element of a single weight tensor. """
+        return self._weight_length
+
+    @property
+    def bias_length(self):
+        """Get the n_element of a single bias tensor. """
+        return self._bias_length
+
+    def set_weight(self, weight, reinitialize=False):
+        """Store a weight tensor or batch of weight tensors."""
```

Comment:
comment the the arguments, especially the shape of weight.

Reply:
updated.